src/datasets/arrow_reader.py

```diff
@@ -105,6 +105,8 @@ def make_file_instructions(name, split_infos, instruction, filetype_suffix=None)
     """
     name2len = {info.name: info.num_examples for info in split_infos}
     if not isinstance(instruction, ReadInstruction):
+        if str(instruction) == "all":
+            instruction = "+".join(name2len.keys())
```

Comment:
we may need to sort the keys here to make the order deterministic

Reply:
`dict` preserves insertion order in Python 3.6+, so the ordering of the keys is deterministic (this is an implementation detail in CPython in 3.6 and was documented in 3.7).